webdictionary/src/main/java/com/eis0/kademlianetwork/informationdeliverymanager/IdFinderHandler.java

```diff
@@ -0,0 +1,184 @@
+package com.eis0.kademlianetwork.informationdeliverymanager;
+
+import androidx.annotation.NonNull;
+
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis0.kademlia.KademliaId;
+import com.eis0.kademlia.SMSKademliaNode;
+import com.eis0.kademlia.SMSKademliaRoutingTable;
+import com.eis0.kademlianetwork.KademliaJoinableNetwork;
+
+import java.util.List;
+
+/**
+ * This class is used to scan the RoutingTable, searching for the {@link SMSKademliaNode} with the
+ * {@link KademliaId} closest to the idToFind (the Resource Id specified in the searchId() call); if
+ * the local node is the closest one it will answer to the searching {@link SMSPeer}, if the closest
+ * node is another node inside of the RoutingTable the local node will pass the task to him (it sends
+ * a SMS to the node, and waits to know if it is still alive inside of the network).
+ * <p>
+ * The searchId() is a wrapper of the searchIdList() method.
+ * <p>
+ * It creates a list of nodes (number of nodes = ATTEMPTS), starting from the closest one; if a node
+ * doesn't respond in 10s the method removes it from the list of closest nodes and restarts the
+ * process with the new list, asking to the following one, and so on, as long as the list ends
+ * or the closest node is the local node. The list length corresponds to the ATTEMPTS value (or to
+ * ATTEMPTS+1, if the list doesn't contain the local node; in this scenario, the local node is added
+ * at the end of the list, as last possible node that will answer to the research).
+ * The nodes are removed only from the list created by the searchIdList, not from the RoutingTable,
+ * to avoid the direct erasure.
+ * <p>
+ * When a node doesn't respond its staleCount get incremented, in the Refresh phase it will be
+ * removed if still inactive.
+ *
+ * @author Marco Cognolato (created the class)
+ * @author Enrico Cestaro (edited and restored the code)
+ * @author Edoardo Raimondi (added the check cicle to verify if the target node is 'alive')
+ */
+public class IdFinderHandler {
+    private static int maxAttempts = 10;
+    private static final String ID_TO_FIND_NULL = "The idToFind cannot be null";
+    private static final String SEARCHER_NULL = "The searcher cannot be null";
+    private static final String NODES_LIST_NULL = "The nodeList cannot be null";
+    private static final String NODES_LIST_EMPTY = "The nodeList cannot be empty";
+private static final String INSUFFICIENT_ATTEMPTS = "The attemptsNumber must be higher then zero";
```

Comment:
Watch out for identation, I see it wrong here.

Reply:
This mistake makes me suffer more than any other